plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -727,22 +742,15 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         }
 
         if (type instanceof TimestampType timestampType) {
-            if (timestampType.getPrecision() <= POSTGRESQL_MAX_SUPPORTED_TIMESTAMP_PRECISION) {
-                verify(timestampType.getPrecision() <= TimestampType.MAX_SHORT_PRECISION);
-                return WriteMapping.longMapping(format("timestamp(%s)", timestampType.getPrecision()), PostgreSqlClient::shortTimestampWriteFunction);
-            }
-            verify(timestampType.getPrecision() > TimestampType.MAX_SHORT_PRECISION);
-            return WriteMapping.objectMapping(format("timestamp(%s)", POSTGRESQL_MAX_SUPPORTED_TIMESTAMP_PRECISION), longTimestampWriteFunction());
+            verify(timestampType.getPrecision() <= TimestampType.MAX_SHORT_PRECISION);
+            return WriteMapping.longMapping(format("timestamp(%s)", timestampType.getPrecision()), PostgreSqlClient::shortTimestampWriteFunction);
         }
         if (type instanceof TimestampWithTimeZoneType timestampWithTimeZoneType) {
-            if (timestampWithTimeZoneType.getPrecision() <= POSTGRESQL_MAX_SUPPORTED_TIMESTAMP_PRECISION) {
-                String dataType = format("timestamptz(%d)", timestampWithTimeZoneType.getPrecision());
-                if (timestampWithTimeZoneType.getPrecision() <= TimestampWithTimeZoneType.MAX_SHORT_PRECISION) {
-                    return WriteMapping.longMapping(dataType, shortTimestampWithTimeZoneWriteFunction());
-                }
-                return WriteMapping.objectMapping(dataType, longTimestampWithTimeZoneWriteFunction());
+            String dataType = format("timestamptz(%d)", timestampWithTimeZoneType.getPrecision());
+            if (timestampWithTimeZoneType.getPrecision() <= TimestampWithTimeZoneType.MAX_SHORT_PRECISION) {
```

Comment:
Can we replace it with `isShort`

Reply:
> Can we replace it with `isShort`

This seems not the same as before: POSTGRESQL_MAX_SUPPORTED_TIMESTAMP_PRECISION = 6